src/routing_table/network_tests.rs

```diff
@@ -17,297 +17,244 @@
-
-// #[test]
-// fn groups_have_identical_routing_tables() {
-//     let mut network = Network::new(None);
-//     for _ in 0..100 {
-//         network.add_node();
-//     }
-//     let keys = network.keys();
-//     for _ in 0..20 {
-//         let address = network.rng.gen();
-//         let close_peer = network.close_node(address);
-//         let mut contacts = network.known_nodes(close_peer);
-//         contacts.sort();
-//         let group = unwrap!(network.nodes[&close_peer].close_nodes(&address));
-//         for &node in &keys {
-//             match network.nodes[&node].close_nodes(&address) {
-//                 None => assert!(!group.contains(&address)),
-//                 Some(nodes) => {
-//                     // TODO: if the node in query itself is part of the close members, there is
-//                     //       chance the group doesn't
-//                     //       got split yet from its perspective
-//                     //       execute this test with seed of [2817869685, 3020312298, 343512997,
-//                     //       915190025]
-//                     //       the fetched group from close_peer is
-//                     //       [7154751357263139303, 7232498110496810657, 7324139913257116470,
-//                     //        7372067402949639454, 7467839279545282818, 7480444072499549176,
-//                     //        7857634283013267913, 7941145200632707472, 8162400431230333977,
-//                     //        8163088974105893084, 8306894308903998305, 8901399224139166674,
-//                     //        8903063204513800509, 8908774849401800370, 9124677007065390519,
-//                     //        9140770996867927707, 9142712693818018265, 9207439687031592853]
-//                     //       which missed itself 7282345220160771589 and the group is not split
-//                     //       yet.
-//                     // Also, `close_nodes` returns the closest known group to the target from our
-//                     // perspective, meanwhile here we are only checking against the
-//                     // `closest_group` assert_eq!(group, nodes);
-//                     let mut result = network.known_nodes(node);
-//                     result.sort();
-//                     assert_eq!(contacts, result);
-//                 }
-//             }
-//         }
-//     }
-// }
-
-// #[test]
-// fn merging_groups() {
-//     let mut network = Network::new(None);
-//     for _ in 0..100 {
-//         network.add_node();
-//         // TODO: Verify invariant.
-//     }
-//     // TODO: Verify that there are several groups.
-//     for _ in 0..95 {
-//         network.drop_node();
-//         // TODO: Verify invariant.
-//     }
-//     // TODO: Verify that there is only one group.
-// }
+#[test]
+fn merging_groups() {
+    let mut network = Network::new(None);
+    for _ in 0..100 {
+        network.add_node();
+        // if i % 5 == 0 {
+        //     verify_invariant(&mut network);
+        // }
```

Comment:
yes, it is currently failing  :(
I remember I have raised this issue somewhere in the slack previously, it is mainly because when we calculate split, node itself is not included, which means you may not split your group, but others did.
with more node populated, this mismatch will gone quickly, however, we do need to double think about it.